plugins/modules/ec2_vpc_net.py

```diff
@@ -616,19 +573,80 @@ def wait_for_updates(connection, module, vpc_id, ipv6_cidr, expected_cidrs, dns_
     if tags is not None:
         tag_list = ansible_dict_to_boto3_tag_list(tags)
         filters = [{"Name": f"tag:{t['Key']}", "Values": [t["Value"]]} for t in tag_list]
-        wait_for_vpc(module, connection, VpcIds=[vpc_id], Filters=filters)
+        wait_for_vpc(module, connection, waiter_name="vpc_available", VpcIds=[vpc_id], Filters=filters)
 
     wait_for_vpc_attribute(connection, module, vpc_id, "enableDnsSupport", dns_support)
     wait_for_vpc_attribute(connection, module, vpc_id, "enableDnsHostnames", dns_hostnames)
 
     if dhcp_id is not None:
         # Wait for DhcpOptionsId to be updated
         filters = [{"Name": "dhcp-options-id", "Values": [dhcp_id]}]
-        wait_for_vpc(module, connection, VpcIds=[vpc_id], Filters=filters)
+        wait_for_vpc(module, connection, waiter_name="vpc_available", VpcIds=[vpc_id], Filters=filters)
 
     return
 
 
+def ensure_present(connection, module: AnsibleAWSModule, vpc_id: str) -> None:
+    name = module.params.get("name")
+    cidr_block = module.params.get("cidr_block")
+    ipv6_cidr = module.params.get("ipv6_cidr")
+    purge_cidrs = module.params.get("purge_cidrs")
+    tenancy = module.params.get("tenancy")
+    dns_support = module.params.get("dns_support")
+    dns_hostnames = module.params.get("dns_hostnames")
+    dhcp_id = module.params.get("dhcp_opts_id")
+    tags = module.params.get("tags")
+    purge_tags = module.params.get("purge_tags")
+
+    if vpc_id is None:
+        if name is None:
+            module.fail_json("The name parameter must be specified when creating a new VPC.")
+        vpc_id = create_vpc_net(connection, module, cidr_block[0], tenancy, tags, ipv6_cidr, name)
```

Comment:
Not sure if feasible, but just a question: What if you make create_vpc_net() return the vpc_obj and use it for update_cirds() instead of making a  get_vpc(...) call?